File: lab4Instrumentos/src/entidades/Instrumento.ts
import { ChangeEventHandler } from "react";
import Categoria from "./Categoria";

export default class Instrumento{
    id:number = 0;
	instrumento:string = "";
	marca:string = "";
	modelo:string = "";
    imagen:string = "";
	precio:number = 0; 
	costoEnvio:string = ""; 
	cantidadVendida:number = 0;
	descripcion:string = "";
	categoria: Categoria = {
		id: 0,
		denominacion: ""
	};
	addCarrito?:ChangeEventHandler;
}

File: Back_Tp_reportes/src/main/java/com/example/Back_Tp_crud/Entity/Instrumento.java
package com.example.Back_Tp_crud.Entity;

import com.example.Back_Tp_crud.Entity.BaseEntity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Instrumento extends BaseEntity {
    private String instrumento;
    private String marca;
    private String modelo;
    private String imagen;
    private Double precio;
    private String costoEnvio;
    private String cantidadVendida;
    private String descripcion;

    // Relación ManyToOne con la entidad Categoria
    @ManyToOne
    @JoinColumn(name = "id_categoria", nullable = false)
    private Categoria categoria;
}
